### ISSUE TITLE
apart drops term

### ISSUE THREAD

**Issue by smichr** [sympy/sympy#9123]
As identified [here](https://groups.google.com/forum/#!topic/sympy/V2qydfXuw7Y) `apart` will sometimes drop a term:

``` python
>>> (x/2).apart()
x/2
>>> (x/2).apart(x)
x/2
>>> (x/2).apart(y)  <------ problem
0
>>> (x/a).apart(y)
x/a
>>> (2*x).apart(y)
2*x
```

**Comment by smichr** [comment]
This happens because of how Poly.quo behaves in the following:

```
>>> from sympy.abc import x, y
>>> Poly(x,x).quo(Poly(2,x))
Poly(1/2*x, x, domain='QQ')
>>> Poly(x,y).quo(Poly(2,y))  <--- problem here: result is 0 but should be x/2
Poly(0, y, domain='ZZ[x]')
```

**Comment by asmeurer** [comment]
@mattpap

**Comment by citizen-seven** [comment]
The issue should be closed, it works correctly now.

**Comment by asmeurer** [comment]
I still get 0 for `(x/2).apart(y)` in master.

**Comment by mohit3011** [comment]
@asmeurer , I would like to work on this issue if this is easy to fix . Could you please refer to any starting point.

**Comment by asmeurer** [comment]
There is some `quo` call in `apart` somewhere that needs a `field=True` parameter. For instance:

```
In [5]: quo(Poly(x,y), Poly(2,y))
Out[5]: Poly(0, y, domain='ZZ[x]')

In [6]: quo(Poly(x,y), Poly(2,y), field=True)
Out[6]: Poly(x/2, y, domain='QQ(x)')
```

**Comment by jksuom** [comment]
#12177 is probably related.

**Comment by jksuom** [comment]
Closed by #12236
